test/functional/legacy/eval_spec.lua

```diff
@@ -495,6 +495,18 @@ describe('eval', function()
       ' abcE4b10-4\000abcE4b10-4-2')
   end)
 
+  it("setreg('0',getreg('a',1,1)) doesn't crash for unset register", function()
```

Comment:
Vim has NULL lists, dictionaries and strings. Hopefully not funcrefs. I do not know a way to produce a NULL dictionary (except for `v:_null_dict`) and did not see crash reports (though most likely there will be some once I start testing), but NULL strings already produced bugs, including the one after which I added `v:_null_…` variables. Since testing that NULL x does not produce a crash is easier then making sure that NULL x does not appear in first place I would suggest to keep them all, but create `test/functional/viml/null_spec.lua` with contents similar to that in #4615.